ftw/publisher/sender/workflows/contextstate.py

```diff
@@ -89,7 +90,7 @@ def is_parent_published(self):
 
     def get_unpublished_references(self):
         for obj in self.get_references():
-            if obj is None:
+            if obj is None or obj == self.context or belongs_to_parent(obj):
```

Comment:
I think `belongs_to_parent(obj)` is now true whenever the reference target is a block.
But should it be true if the target is a block of another page which is not published?

Reply:
Then I shouldn't use `belongs_to_parent` and replace the `belongs_to_parent` with the stuff before. Or what would you do?

Reply:
Because `belongs_to_parent` only checks if the obj is of a certain interface and if it has a workflow. It does know anything about its parent and we have to check for the parent.

Reply:
I think the point is that you do not want to check blocks, but always check the container which has a workflow.

So basically you could something like this:

```python
for obj in self.get_references():
  if obj is None:
    continue

  # When obj is a block we want to check the status on the parent container
  # which has a status.
  # This is done by walking up in the chain finding the first object which
  # does not belong to the parent; this might be the `obj` itslef.
  for obj_or_parent in aq_chain(aq_inner(obj)):
    if not belongs_to_parent(obj_or_parent):
        obj = obj_or_parent
        break

  ...
```

Reply:
I have made the discussed changes